app/Platform/Actions/ComputeGameSearchTitlesAction.php

```diff
@@ -0,0 +1,387 @@
+        'VI' => '6', 'VII' => '7', 'VIII' => '8', 'IX' => '9', 'X' => '10',
+        'XI' => '11', 'XII' => '12', 'XIII' => '13', 'XIV' => '14', 'XV' => '15',
+        'XVI' => '16', 'XVII' => '17', 'XVIII' => '18', 'XIX' => '19', 'XX' => '20',
+    ];
+
+    public function execute(string $gameTitle, array $altTitles = []): array
+    {
+        $searchTitles = [];
+
+        // Process the main title first.
+        $mainTitleVariations = $this->generateSearchVariationsForTitle($gameTitle);
+        $searchTitles = array_merge($searchTitles, $mainTitleVariations);
+
+        // Then, process each alternative title.
+        foreach ($altTitles as $altTitle) {
+            $altTitleVariations = $this->generateSearchVariationsForTitle($altTitle);
+            $searchTitles = array_merge($searchTitles, $altTitleVariations);
+        }
+
+        // Remove duplicates and empty values to keep the index clean.
+        $searchTitles = array_values(array_unique(array_filter($searchTitles)));
+
+        return $searchTitles;
+    }
+
+    /**
+     * Generate all search variations for a single title.
+     *
+     * This creates multiple versions of the title to handle common search patterns,
+     * including case variations, numeral conversions, abbreviations, and simplified forms.
+     */
+    private function generateSearchVariationsForTitle(string $title): array
+    {
+        $variations = [];
+
+        // Always include the original title as-is.
+        $variations[] = $title;
+
+        // Add lowercase for case-insensitive matching.
+        $variations[] = mb_strtolower($title);
+
+        // Convert Roman numerals to numbers for searches like "final fantasy 7".
+        $titleWithNumbers = $this->convertRomanNumeralsToNumbers($title);
+        if ($titleWithNumbers !== $title) {
+            $variations[] = $titleWithNumbers;
+            $variations[] = mb_strtolower($titleWithNumbers);
+        }
+
+        // Convert numbers to Roman numerals for searches like "final fantasy vii".
+        $titleWithRomanNumerals = $this->convertNumbersToRomanNumerals($title);
+        if ($titleWithRomanNumerals !== $title) {
+            $variations[] = $titleWithRomanNumerals;
+            $variations[] = mb_strtolower($titleWithRomanNumerals);
+        }
+
+        // Generate common abbreviations like "ff7" or "sm64".
+        $abbreviations = $this->generateAbbreviations($title);
+        foreach ($abbreviations as $abbreviation) {
+            $variations[] = $abbreviation;
+        }
+
+        // Extract series names for partial matching like "zelda" or "mario".
+        $seriesNames = $this->extractSeriesNames($title);
+        foreach ($seriesNames as $seriesName) {
+            $variations[] = $seriesName;
+        }
+
+        // Simple check for Professor Oak Challenge subset.
+        if (stripos($title, 'Professor Oak Challenge') !== false) {
+            $variations[] = 'poc';
+        }
```

Comment:
Another case where it would be beneficial to just allow an admin to add custom aliases

Reply:
See: https://github.com/RetroAchievements/RAWeb/pull/3552#discussion_r2112866176

POC sets are especially egregious when it would come to DB maintenance / human intervention over game records, as these are very frequently created. I think handling it in the code here is actually pragmatic, as it lowers the human cost to just the cost of maintaining this conditional.